### PR TITLE
init commit, implementation plus unit test

### PR DESCRIPTION
**Pull request by alexmengyuan**
test covered encrypt a content, with receiver publickey , then decrypt using receiver seed, seed and publickey pair is generated from iota ed25519 class.

### CODE REVIEW

File: src/aes.test.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+describe('basic test for ecies ed25519',()=>{
+    let receiverInfo:{secret:Uint8Array,publicKey:Uint8Array}
+    let senderInfo:{secret:Uint8Array,publicKey:Uint8Array}
+    const basicContent = 'hehe'
+    const basicAesKey = '12321321321321312'
+    const tag = 'IOTACAT'
+    beforeEach(()=>{
+        receiverInfo = getEphemeralSecretAndPublicKey()
+        senderInfo = getEphemeralSecretAndPublicKey()
+    })
```
Comment (tanglepay-dev) [inline src/aes.test.ts]: unused?

File: src/edwards25519/bigIntCommon.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 IOTA Stiftung
```
Comment (tanglepay-dev) [inline src/edwards25519/bigIntCommon.ts]: seems iota/crypto-js is already referenced by the package. is it still needed to copy over the edwards25519 folder?

File: src/aes.test.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { aesEncrypt, aesDecrypt, getKeyAndIv, getEphemeralSecretAndPublicKey, productOfTwo } from '.';
+import { Bip39 } from '@iota/crypto.js'
+import CryptoJS from 'crypto-js';
+
+describe('basic test for ecies ed25519',()=>{
+    let receiverInfo:{secret:Uint8Array,publicKey:Uint8Array}
+    let senderInfo:{secret:Uint8Array,publicKey:Uint8Array}
+    const basicContent = 'hehe'
+    const basicAesKey = '12321321321321312'
+    const tag = 'IOTACAT'
```
Comment (tanglepay-dev) [inline src/aes.test.ts]: better to use a dummy tag name

File: src/encrypt.test.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    let decryptResult:{
+        ephemeralPublicKey:Uint8Array,
+        aesKey:string,
+        encrypted:string,
+        payload:string
+    }
+    const tag = 'IOTACAT'
+    beforeEach(()=>{
+        receiverInfo = getEphemeralSecretAndPublicKey()
+        contentToBeEncrypted = 'hehe'//Bip39.randomMnemonic(128)
```
Comment (tanglepay-dev) [inline src/encrypt.test.ts]: better to test some chinese text as well to make sure byte encodings are handled correctly

File: src/edwards25519/bigIntCommon.ts (hunk 2 above)
Comment (tanglepay-dev) [reply src/edwards25519/bigIntCommon.ts]: likewise for the utils folder

File: src/aes.test.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+
```
Comment (tanglepay-dev) [inline src/aes.test.ts]: better to move .test.ts files to a different folder

File: src/index.ts [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    sha512.update(bytes);
+
+    const digest = sha512.digest();
+    digest[0] &= 248;
+    digest[31] &= 127;
+    digest[31] |= 64;
+    return digest.slice(0,32);
+}
+
+export const  getEphemeralSecretAndPublicKey = () => {
```
Comment (tanglepay-dev) [inline src/index.ts]: better to move test utilities into test folder

File: src/index.ts [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    const sha512 = new Sha512();
+    sha512.update(bytes);
+
+    const digest = sha512.digest();
+    digest[0] &= 248;
+    digest[31] &= 127;
+    digest[31] |= 64;
+    return digest.slice(0,32);
+}
+
```
Comment (tanglepay-dev) [inline src/index.ts:43]: would it be more easier to use if we provide interfaces that takes target ed25519 addresses directly?

File: src/aes.test.ts (hunk 1 above)
Comment (alexmengyuan) [reply src/aes.test.ts]: will remove

File: src/aes.test.ts (hunk 3 above)
Comment (alexmengyuan) [reply src/aes.test.ts]: will change to 'tag'

File: src/edwards25519/bigIntCommon.ts (hunk 2 above)
Comment (alexmengyuan) [reply src/edwards25519/bigIntCommon.ts]: class ExtendedGroupElement which is used during testing interoperability of iota and ExtendedPoint of noble ed2551
 is not exported from @iota/crypto.js at the moment, could remove that one test though, other test that testing publicKey equals product of base point and scalar from third party's implementation could also proves interoperability I guess

**Comment by alexmengyuan** [comment]
updated based on comments

**Review by tanglepay-dev — DISMISSED** [review]

**Review by tanglepay-dev — APPROVED** [review]

**Pull request merged**